src/main.cpp

```diff
@@ -1622,12 +1622,16 @@ void CheckForkWarningConditions()
     if (IsInitialBlockDownload())
         return;
 
+    const CBlockIndex* pChainTip = GetChainTip();
```

Comment:
This one is not needed, it's a recursive lock. At this point, we already ensured that `cs_main` is locked. We can access `chainActive` directly.

Reply:
The point of `GetChainTip` is  not only locking `cs_main`, but returning a pointer in `mapBlockIndex` rather than `chainActive`. Agreed though, that we should avoid recursive locks whenever possible.
Can either restore `chainActive.Tip()`, or make a non-locking version of `GetChainTip` (called by the other one).

Reply:
yeah ok.
A good convention to follow could be:
* If the method is entirely under `cs_main` lock, then can access directly to `chainActive` (no need to look in `mapBlockIndex`, it will only add overhead without any benefit).
* If we only need the tip `CBlockIndex*` to perform some calculation and we are not locking `cs_main` over the entire function, then we call `GetChainTip()`.